utilities/deep_inspection.py

```
@@ -550,3 +616,102 @@ def wait_for_di_snapshot_cleanup(
+    fixture_store.pop(DI_RESULTS_KEY, None)
+    last_poll_time = 0.0
+    poll_interval = 10
+    captured_vm_names: set[str] = set()
+    was_executing = False
+
+    def _try_capture() -> None:
+        """Query DI CRs and capture any newly-Succeeded results."""
+        try:
+            conversions = get_plan_conversion_crs(plan_resource=plan, conversion_type=CONVERSION_TYPE_DEEP_INSPECTION)
+        except NotFoundError:
+            LOGGER.debug(f"Plan or Conversion CR not found during DI capture for plan '{plan.name}'")
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)